### PR TITLE
PL-1 : Refactor code and use material design

### PR DESCRIPTION
**Pull request by nirajcj**

### CODE REVIEW

File: app/src/main/res/xml/normal_hara_button_selector.xml [hunk 1]
```diff
@@ -1,7 +1,9 @@
 <?xml version="1.0" encoding="utf-8"?>
   <selector xmlns:android="http://schemas.android.com/apk/res/android">
-  <item android:state_pressed="true" 
-    android:drawable="@xml/mytextbox_pressed" />
-  <item  android:state_focused="false" 
-    android:drawable="@xml/mytextbox" />
-  </selector>
\ No newline at end of file
+  <item
```
Comment (thesilentpsycho) [inline app/src/main/res/xml/normal_hara_button_selector.xml]: hara button selector :D

File: app/src/main/java/com/upa/passwordlocker/utils/CommonUtils.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        if(data == null) {
+            return false;
+        }
+
+        return data.contains(" ");
+    }
+
+    public static boolean hasActivityNotStarted(Context context, Intent intent) {
+        try {
+            context.startActivity(intent);
```
Comment (thesilentpsycho) [inline app/src/main/java/com/upa/passwordlocker/utils/CommonUtils.java:20]: ye hasActivityNotStarted method ko activity start krna chahiye kya ? Isko bola to nahi gya hai start krne. Isse pucha gaya hai ?

File: app/src/main/java/com/upa/passwordlocker/utils/CustomDbHelper.java [hunk 3]
```diff
@@ -75,7 +78,7 @@ public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
 						arr[j + 2]=AESHelper.encrypt(arr[j + 2]);
 					}
 					catch (Exception e) {
-
```
Comment (thesilentpsycho) [inline app/src/main/java/com/upa/passwordlocker/utils/CustomDbHelper.java]: e.getMessage() bhi log kr do. Stacktrace bhi

File: app/src/main/java/com/upa/passwordlocker/views/SqlDataViewActivity.java [hunk 4]
```diff
@@ -231,41 +243,41 @@ else if(item.getTitle().toString().equals("Delete")) {
...
-			int k = 0;
-			groups.clear();
+		int k = 0;
+		groups.clear();
 
-			CustomDbHelper info = new CustomDbHelper(this);
-			info.open();
-			String data = info.getData();
-			info.close();
+		CustomDbHelper info = new CustomDbHelper(this);
```
Comment (thesilentpsycho) [inline app/src/main/java/com/upa/passwordlocker/views/SqlDataViewActivity.java]: rename info

File: app/src/main/java/com/upa/passwordlocker/views/SqlDataViewActivity.java [hunk 5]
```diff
@@ -231,41 +243,41 @@ else if(item.getTitle().toString().equals("Delete")) {
...
+		info.open();
```
Comment (thesilentpsycho) [inline app/src/main/java/com/upa/passwordlocker/views/SqlDataViewActivity.java]: yahan try lagana hoga shayad

File: app/src/main/java/com/upa/passwordlocker/views/SqlDataViewActivity.java [hunk 6]
```diff
@@ -231,41 +243,41 @@ else if(item.getTitle().toString().equals("Delete")) {
...
+		String data = info.getData();
+		info.close();
 
-			String username_decrypted;
-			String password_decrypted;
+		String username_decrypted;
+		String password_decrypted;
 
-			String[] arr = data.split("\n");
+		String[] arr = data.split("\n");
```
Comment (thesilentpsycho) [inline app/src/main/java/com/upa/passwordlocker/views/SqlDataViewActivity.java:258]: null pointer

File: app/src/main/java/com/upa/passwordlocker/utils/CustomDbHelper.java [hunk 7]
```diff
@@ -47,7 +45,7 @@ public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
 
 			if(oldVersion == 1) {
```
Comment (thesilentpsycho) [inline app/src/main/java/com/upa/passwordlocker/utils/CustomDbHelper.java:49]: Ye old version wala kuch hack hai us time ka. deployment pe kuch phat gya tha so to support existing 30-40 users we had to add it. Isko ab hmlog hata skte hain. dekh lena.

**Review by thesilentpsycho — APPROVED** [review]

File: app/src/main/java/com/upa/passwordlocker/views/ChangePinActivity.java [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+import android.view.View;
+import android.view.View.OnClickListener;
+import android.widget.Toast;
+
+import com.google.android.material.button.MaterialButton;
+import com.google.android.material.textfield.TextInputEditText;
+import com.upa.passwordlocker.R;
+import com.upa.passwordlocker.utils.EncryptionHelper;
+
+public class ChangePinActivity extends Activity implements OnClickListener{
```
Comment (utkbits) [inline app/src/main/java/com/upa/passwordlocker/views/ChangePinActivity.java]: OnClickListener ke baad space daal sakte :P yahi comment soch paya 🗡️

**Review by utkbits — APPROVED** [review]
mere ko kuch galat bhi milega to maan ke chalunga bhagbo sahi kiya hoga

**Pull request merged**